src/averageAges.js

```diff
@@ -15,11 +15,11 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with logical operators (&&, ||) or ternary operator (?:)
-  // without nesting
+  return people.filter(person =>
+    (person.sex === 'm')
+      && (!century || (century && (Math.ceil(person.died / 100)) === century)))
```

Comment:
too hard to read, understand and maintain

Reply:
> too hard to read, understand and maintain

refactored